overlord/ifacestate/helpers.go

```diff
@@ -98,6 +101,24 @@ func (m *InterfaceManager) addSnaps() error {
 	return nil
 }
 
+// fixDisconnectedCorePlugs will re-connect the network-bind plug to the
+// network-bind slot (on core). This cures the effects of bug
+// https://bugs.launchpad.net/snappy/+bug/1680097
+func (m *InterfaceManager) fixDisconnectedCorePlugs() error {
+	// If the core snap has a network-bind plug
+	if m.repo.Plug("core", "network-bind") != nil {
+		// connect it to the slot (connect is a no-op if connected)
+		connRef := interfaces.ConnRef{
+			PlugRef: interfaces.PlugRef{Snap: "core", Name: "network-bind"},
```

Comment:
This looks wrong and needs fixing right away. I have no idea how we allowed duplicated slot+plug names, but that's part of the foundation of the system. We cannot allow that to happen at all.